src/protocol/RedemptionHandler.sol

```diff
@@ -63,6 +69,15 @@ contract RedemptionHandler is CoreOwnable{
         emit SetDiscountInfo(_rate, _maxUsage, _maxDiscount);
     }
 
+    function setOverusageInfo(uint256 _rate, uint256 _start, uint256 _end) external onlyOwner{
+        require(_rate <= baseRedemptionFee*5, "over usage fee too high");
+        require(_start <= _end && _end <= 10_000, "invalid start/end");
```

Comment:
if start is `ever == end`, could revert with divide by zero on line 148
`require(_start < _end && _end <= 10_000, "invalid start/end");`